low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -32,18 +44,125 @@ class ServerMap(ResourceMap):
 
     def __init__(self):
         """Initialize server"""
-        pass
+        self.server_map = {
+            "memory": self.get_mem_info
+            }
 
     def get_health_info(self, rpath):
         """
         Get health information of fru in given resource id
 
         rpath: Resource id (Example: nodes[0]>compute[0]>hw>disks)
         """
-        pass
+        info = {}
+        nodes = rpath.strip().split(">")
+        leaf_node, _ = self.get_node_details(nodes[-1])
+        if leaf_node == "compute":
+            info["last_updated"] = int(time.time())
+            for fru in self.server_map:
+                info.update({fru: self.server_map[fru]()})
+        else:
+            fru = None
+            fru_found = False
+            for node in nodes:
+                fru, _ = self.get_node_details(node)
+                if self.server_map.get(fru):
+                    fru_found = True
+                    info = self.server_map[fru]()
+                    break
+        if not fru_found:
+            raise ResourceMapError(
+                errno.EINVAL,
+                f"Health provider doesn't have support for'{rpath}'.")
+        return info
+
+    def get_mem_info(self):
+        """Update and return memory info in specific format"""
```

Comment:
Remove blank as mentioned by Codacy report.

Reply:
Removed get_mem_info() as no need to read the cache mem data